Move large units to the side

Iridar opened the pull request:
FIxes #84

Move SPARKs and other large units slightly to the right so they no longer obscure the promotion screen.

Iridar commented:
Brigadier rank:

Before: https://i.imgur.com/xRzmCry.jpg

After: https://i.imgur.com/Y4pLLNE.jpg

No brigadier:

Before: https://i.imgur.com/m5TmDJx.jpg

After: https://i.imgur.com/SRGQ7Ll.jpg

robojumper commented on X2WOTCCommunityPromotionScreen/Src/X2WOTCCommunityPromotionScreen/Classes/CPS_UIArmory_PromotionHero.uc:
```diff
@@ -229,6 +233,24 @@ simulated function PopulateData()
+		{
+			PawnLocation.X += LargeUnitPawnOffset;
+		}
+		XComUnitPawnNativeBase(ActorPawn).SetLocationNoCollisionCheck(PawnLocation);
```
I'm not 100% sure if this is the correct place to update this -- conceptually this should probably happen when actually creating the pawn so that the `UIPawnMgr`'s internal info also has the correct info.

Also this makes `PopulateData` not idempotent, which seems dangerous to me -- if `PopulateData` is called multiple times (skimming the code, e.g. by purchasing more abilities) the Spark would move a bunch of units to +X multiple times...

Iridar replied:
> I'm not 100% sure if this is the correct place to update this -- conceptually this should probably happen when actually creating the pawn so that the `UIPawnMgr`'s internal info also has the correct info.
> 
> Also this makes `PopulateData` not idempotent, which seems dangerous to me -- if `PopulateData` is called multiple times (skimming the code, e.g. by purchasing more abilities) the Spark would move a bunch of units to +X multiple times...

Good points, will adjust.

Iridar commented:
Ran some tests, and you were right, the pawn was indeed getting slowly right-shifted into oblivion when entering and exiting the promotion screen repeatedly.

I've adjusted the code to add the offset to the location we get from the armory's placement actor instead of the pawn itself. Since the location of the placement actor remains unchanged, the pawn won't get offset from it more than once.

I've also moved the offset function call from `PopulateData()` into `CreateSoldierPawn()`, though due to Firaxis' noodle code, that function doesn't actually release the old pawn, and neither does the `RequestPawn()`, which is the function that's actually responsible for creating the pawn... most of the time. The underlying code is kinda awful, so there's no single good place to put the offset function call into. 

I've also added some code to move the pawn back to the placement actor's location when the promotion screen is removed.

I briefly tested the new version and everything seems to be working fine.
